From: coopi <coldsideofyourpillow@disroot.org>
Date: Sat, 22 Aug 2026 10:39:00 +0400
Subject: [PATCH] futur-tests.el: Skip broken pre-Emacs-31 interruption test

Futur interrupts Elisp servers on Emacs before 31 by sending SIGUSR1 and using
`debug-on-event' to turn the signal into a `quit'.  On Emacs 30.2, that path
enters the debugger instead, leaving the server busy and causing the
interruption test to fail.

Use the same `kill-emacs-on-sigint' capability check as Futur itself to skip
only the known-broken interruption test.  Keep the rest of
`futur--tests-elisp-funcall' enabled.
---
 futur-tests.el | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/futur-tests.el b/futur-tests.el
--- a/futur-tests.el
+++ b/futur-tests.el
@@ -279,5 +279,6 @@
-  (if (eq elisp-funcall #'futur-elisp-sandbox-funcall)
-      (message "Skipping interruption test, known to fail in sandboxes")
+  (if (or (eq elisp-funcall #'futur-elisp-sandbox-funcall)
+          (not (boundp 'kill-emacs-on-sigint)))
+      (message "Skipping interruption test, known to fail in sandboxes or before Emacs 31")
     (defvar futur--last-result)
     (let ((inhibit-interaction nil)
           ;; (start (float-time))
-- 
2.51.0
